### ISSUE TITLE
Iterable type params are not checked for function params

### ISSUE THREAD

**Issue by weirdan** [vimeo/psalm#1359]
https://getpsalm.org/r/6ccdab52e3
```php
<?php
  /** @param iterable<string,object> $_p */
  function accepts(iterable $_p): void {}

  /** @return iterable<int,int> */
  function iterable() { yield 1; }

  /** @return Traversable<int,int> */
  function traversable() { yield 1; }

  /** @return Generator<int,int,mixed,void> */
  function generator() { yield 1; }

  /** @return array<int,int> */
  function arr() { return [1]; }

  accepts(iterable());
  accepts(traversable());
  accepts(generator());
  accepts(arr());
```
**Expected**: All calls are flagged
**Actual**: Only `accepts(arr())` call is flagged
```
Psalm output (using commit 3fdef1a): 

ERROR: InvalidArgument - 20:11 - Argument 1 of accepts expects iterable<string, object>, array<int, int> provided
```
